### PR TITLE
feat: support multiple table readers

### PR DESCRIPTION
**Pull request by egorchakov**
- add support for multiple table readers
- add a `protobuf -> pl.DataFrame` mcap decoder factory
- replace `CarlaRecordsTableBuilder` with `JsonTableReader`

### CODE REVIEW

File: examples/config_templates/table_builder/yaak.yaml [hunk 1]
```diff
@@ -7,38 +7,53 @@
...
-        _target_: polars.Enum
-        categories: ["0", "1", "2", "3"]
+          speed: polars.Float32
+          gear:
+            _target_: polars.Enum
+            categories: ["0", "1", "2", "3"]
+
+  - path: ???
+    reader:
+      _target_: rbyte.io.table.mcap.McapTableReader
```
Comment (sandhawalia) [inline examples/config_templates/table_builder/yaak.yaml:47]: ❤️

**Comment by sandhawalia** [comment]
LGTM. Maybe we overlooked, after adding support for video decoding (CPU/GPU), we didn't add a sample dataset config which uses video CPU / GPU decoder. Both `yaak.yaml/carla.yaml` have `DirectoryFrameReader`. 

Does it make sense to add atleast one of these ? Or convert one of the existing one to use HW accelerated video decoding ?

```
examples/config_templates/dataset/yaak.gpu.yaml
examples/config_templates/dataset/carla.gpu.yaml
```

File: examples/config_templates/dataset/yaak.yaml [hunk 2]
```diff
@@ -1,7 +1,7 @@
 #@yaml/text-templated-strings
 
 #@ drives = [
-#@     'Niro098-HQ/2024-08-26--06-06-03',
+#@     'Niro098-HQ/2024-01-22--09-03-16',
```
[the remark this answers is not in the document]
Comment (egorchakov) [reply examples/config_templates/dataset/yaak.yaml:4]: picked this one b/c it has `ai.mcap`

**Comment by egorchakov** [comment]
@sandhawalia 
> Maybe we overlooked, after adding support for video decoding (CPU/GPU), we didn't add a sample dataset config which uses video CPU / GPU decoder. Both yaak.yaml/carla.yaml have DirectoryFrameReader.

will address once we move examples to notebooks

**Pull request merged**